proxy/proxy.go

```diff
@@ -1045,6 +1046,10 @@ func (p *Proxy) do(ctx *context) error {
 		p.log.Debugf("deprecated shunting detected in route: %s", ctx.route.Id)
 		return &proxyError{handled: true}
 	} else if ctx.shunted() || ctx.route.Shunt || ctx.route.BackendType == eskip.ShuntBackend {
+		// consume the body to prevent goroutine leaks
+		if ctx.request.Body != nil {
+			io.Copy(ioutil.Discard, ctx.request.Body)
```

Comment:
body should be closed, too.

Reply:
Good catch!

Reply:
actually, no :)

See the docs for the Body field of the Request type: https://pkg.go.dev/net/http/?tab=doc#Request